stackrox/acs-fleet-manager
# ROX-15941: Run multiple operators from FSS

## Description
Make FSS able to run multiple ACS Operator versions.
Each operator version has its own deployment.
The metrics service and ServiceAccount can be shared between multiple versions.
For the sake of simplicity, there are two versions installed on startup before reconciler run.

The next steps are:
- fetch operators versions from FM
- installed desired versions on ticker base manner (similar to centrals reconcile)
- cleanup redundant operator versions

## Checklist (Definition of Done)


## Test manual

First of all, the OLM based ACS Operator should be deleted without removing an existing CRDs
1. List and delete OLM subscription for the ACS Operator

```
kubectl get subscription -n stackrox-operator
kubectl delete subscription <subscription> -n stackrox-operator
```

2. Delete the Operator’s ClusterServiceVersion

```
kubectl get clusterserviceversion -n stackrox-operator
kubectl delete clusterserviceversion rhacs-operator.<version> -n stackrox-operator
```

3. Check that there is no running ACS Operator

```
kubectl get pods -n stackrox-operator
NAME                                                              READY   STATUS      RESTARTS      AGE
```

Then it's time to run and verify
4. Build and deploy

```
make deploy/dev
```

5.  Check that a two ACS Operator versions are running

```
kubectl get pods -n stackrox-operator
NAME                                                              READY   STATUS      RESTARTS       AGE
c8dae07dab84cfc9348095952222435b18a827136292688d44a0b210f6nf8dl   0/1     Completed   0              118d
rhacs-operator-controller-manager-3.74.0-5fb9cbcdbb-56nqv         2/2     Running     0              13s
rhacs-operator-controller-manager-3.74.1-5765676ffc-l9bpp         2/2     Running     0              13s
...
```

5.  Check deployments

```
kubectl get deployments -n stackrox-operator
NAME                                       READY   UP-TO-DATE   AVAILABLE   AGE
rhacs-operator-controller-manager-3.74.0   1/1     1            1           27s
rhacs-operator-controller-manager-3.74.1   1/1     1            1           27s
```
